_api-reference/ingest-apis/ip2geo.md

```diff
@@ -0,0 +1,212 @@
+
+If OpenSearch cannot update a datasource from those three endpoints in 30 days, OpenSearch does not add GeoIP data to documents, instead it adds `"error":"ip2geo_data_expired"`.
+
+The following table describes the IP2geo datasource options.
+
+| Name | Required | Default | Description |
+|------|----------|---------|-------------|
+| endpoint | no | https://geoip.maps.opensearch.org/v1/geolite2-city/manifest.json | The endpoint to download GeoIP data|
+| update_interval_in_days | no | 3 | The frequency in days to update GeoIP data with minimum value as 1 |
+
+The following code example shows how to create a Ip2Geo datasource.
+
+#### Example: Request
+
+````json
+PUT /_plugins/geospatial/ip2geo/datasource/my-datasource
+{
+    "endpoint" : "https://geoip.maps.opensearch.org/v1/geolite2-city/manifest.json",
+    "update_interval_in_days" : 3
+}
+```
+
+The following code example shows the JSON reponse to the preceding request. A true JSON response means the request was successful and the server was able to process the request. If you receive a false JSON reponse, check the request to make sure it is valid, check the URL to make sure it is correct, or try again.
+
+#### Example: JSON response
+
+```json
+{
+    "acknowledged":true
+}
+```
+
+## Sending a GET request
+
+To get information about one or more IP2geo datasource, send a GET request.  
+
+#### Example: Request
+
+```
+GET /_plugins/geospatial/ip2geo/datasource/my-datasource
+```
+
+#### Example: Response
+
+```
+{
+  "datasources": [
+    {
+      "name": "my-datasource",
+      "state": "AVAILABLE",
+      "endpoint": "https://geoip.maps.opensearch.org/v1/geolite2-city/manifest.json",
+      "update_interval_in_days": 3,
+      "next_update_at_in_epoch_millis": 1685125612373,
+      "database": {
+        "provider": "maxmind",
+        "sha256_hash": "0SmTZgtTRjWa5lXR+XFCqrZcT495jL5XUcJlpMj0uEA=",
+        "updated_at_in_epoch_millis": 1684429230000,
+        "valid_for_in_days": 30,
+        "fields": [
+          "country_iso_code",
+          "country_name",
+          "continent_name",
+          "region_iso_code",
+          "region_name",
+          "city_name",
+          "time_zone",
+          "location"
+        ]
+      },
+      "update_stats": {
+        "last_succeeded_at_in_epoch_millis": 1684866730192,
+        "last_processing_time_in_millis": 317640,
+        "last_failed_at_in_epoch_millis": 1684866730492,
+        "last_skipped_at_in_epoch_millis": 1684866730292
+      }
+    }
+  ]
+}
+```
+
+## Updating IP2geo datasource
+
+Update IP2geo datasource. The GeoIP data from the new endpoint should contain all fields in GeoIP data from the current endpoint for the update to succeed.
+
+#### Example: Request
+
+```json
+{
+    "endpoint": https://geoip.maps.opensearch.org/v1/geolite2-city/manifest.json,
+    "update_interval_in_days": 10
+}
+```
+
+#### Example: Response
+
+```json
+{
+    "acknowledged":true
+}
+```
+
```

Comment:
Add Deleting IP2geo datasource here

## Deleting IP2geo datasource
Delete GeoIP datasource. Note that if you have processors that use the datasource, the delete requests will fail. To delete the datasource, you must delete all processors associated with the datasource first. 

#### Example: DELETE request
```json
DELETE /_plugins/geospatial/ip2geo/datasource/my-datasource
```

#### Example: response
```json
{
  "acknowledged": true
}
```